src/window_identifier.rs

```diff
@@ -241,11 +246,157 @@ impl WindowIdentifier {
         match handle {
             Some(h) => WindowIdentifier::Gtk3 {
                 handle: h,
-                window: Arc::new(Mutex::new(win.clone().upcast())),
+                window: Arc::new(Mutex::new(Some(win.clone().upcast()))),
             },
             None => WindowIdentifier::default(),
         }
     }
+
+    #[cfg(feature = "raw_handle")]
+    /// Create an instance of [`WindowIdentifier`] from a [`RawWindowHandle`](raw_window_handle::RawWindowHandle`).
+    ///
+    /// # Panics
+    ///
+    /// If the passed RawWindowHandle is not of `WaylandHandle` or `XlibHandle` type. As other platforms handles
+    /// are not supported by the portals so far.
+    pub async fn form_raw_handle(handle: RawWindowHandle) -> Self {
+        use raw_window_handle::RawWindowHandle::{Wayland, Xlib};
+        match handle {
+            // TODO: figure out how to make use of the wl_display to export a handle with xdg-foreign protocol
+            Wayland(_wl_handle) => Self::default(),
```

Comment:
cc @Be-ing, this is sub-optimal but I have no idea how to do this using the wayland-client crate. It seems like we could make use of https://smithay.github.io/wayland-rs/wayland_protocols/unstable/xdg_foreign/v2/client/zxdg_exporter_v2/struct.ZxdgExporterV2.html but then I can't figure out how to get an instance of `ConnectionHandle`

Reply:
I don't intend to spend any more time on implementing this part, so someone else will have to deal with that.

Reply:
This will be problematic, especially after the migration of the wayland-rs ecosystem to 0.30, as ConnectionHandle is something that user owns, so the only way is to pass it down.

@vberger Sorry for a ping, but do you perhaps have any thoughts on this? For context, we are trying to export a toplevel for use in XDG Portal, but raw-window-handle only provides two raw pointers to `surface` and `display`, so we have no way to create resources or sent any requests to the server. My initial idea is to just pass conn to it, like we do in Smithay and sctl, but having a Wayland specific function kinda defeats the purpose or `raw_window_handle`

Reply:
@PolyMeilex the code already makes use of 0.30  btw

Reply:
Yep, I meant that ecosystem is not ported yet, like `sctl` and `winit`